app/models/user.rb

```diff
@@ -3,8 +3,16 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  
+         
+  has_many :chat_rooms, dependent: :destroy
+  has_many :messages, dependent: :destroy
 
   def create
 
   end
+
+  def name
+    email.split('@')[0]
```

Comment:
It's not required, but you can also use `email.split('@').first` which reads slightly better.